oncodash-app/docs/sphinx-js/source/composition.rst

```diff
@@ -0,0 +1,358 @@
+                  <div @selected=${this.onSelected}>
+                      <slot name="selector" />
+                  </div>
+                  <div>
+                      <slot name="viewer" />
+                  </div>`;
+        }
+
+        // Called when our `selected` event bubbles up
+        // from the Selector widget.
+        private onSelected(e : CustomEvent) {
+            console.log("[Mediator] Received selected from selector: ",e.detail.id);
+            // The event embbeds the selected `item_id`.
+            // We update the property, which will automagically
+            // trigger an update sequence.
+            this.item_id = e.detail.id;
+        }
+
+        // This is a handle on the widget that's in the Viewer slot.
+        // It return an array, and the child widget is in the first element.
+        @queryAssignedElements({slot:"viewer"})
+        viewer!: Array<HTMLElement>;
+
+        // The corresponding getter.
+        getViewer(): HTMLElement {
+            return this.viewer[0];
+        }
+
+        // Every time the Mediator is updated
+        //   (whether it's because `this.item_id` was changed directly,
+        //   or because it received an event from a child —selector— widget),
+        // we propagate the `item_id` down to the widget in the Viewer slot.
+        override updated(changedProperties:PropertyValues<any>): void {
+            // Propagate the update to the Lit super class.
+            super.updated(changedProperties);
+
+            if(!Number.isNaN(this.item_id)) { // Sanity check.
+                // Handle of the widget itself.
+                let child = this.getViewer();
+                console.log("[Mediator] Set child viewer widget's selection to: ", this.item_id);
+                // Change the item_id attribute of the Viewer widget.
+                child.setAttribute("item_id", `${this.item_id}`);
+            }
+        }
+    }
+
+    declare global {
+      interface HTMLElementTagNameMap {
+        'my-mediator': Mediator;
+      }
+    }
+
+
+Viewer Widget
+=============
+
+.. code-block:: typescript
+    import {LitElement, html} from 'lit';
+    import {customElement, property, state} from 'lit/decorators.js';
+
+    // The Viewer widget is responsible for displaying any information
+    // that comes down from the Mediator.
+    @customElement('my-viewer')
+    export class Viewer extends LitElement {
+
+        // We overload the setter for the `item_id` attribute
+        // because we want to trigger an update of the corresponding data.
+        // Hence, every time the Mediator widget does change this attribute,
+        // we will fetch new data.
+        @property({type: Number})
+        set item_id(item_id: number) {
+            console.log("[Viewer] Set item_id to", item_id);
+            // The "true" attribute is private and prefixed by convention.
+            this._item_id = item_id;
+            this.fetchItem(item_id);
```

Comment:
Ah, okay. I haven't used reactive properties or `@state`. If I understand correctly, the reactive properties are observed for changes and automatically request update when changed. Thus, an explicit `requestUpdate` isn't even needed (?).

Reply:
That's what I understood.

Reply:
I (once again) failed to manage the asynchronous stuff around making the set of both variables atomic.
I will not loose more time on this, but I set issue #56 with high priority, in case someone would be interested.